src/qoi/src/rayfire_mesh.C

```diff
@@ -0,0 +1,331 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// GRINS - General Reacting Incompressible Navier-Stokes
+//
+// Copyright (C) 2014-2016 Paul T. Bauman, Roy H. Stogner
+// Copyright (C) 2010-2013 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+
+// This class
+#include "grins/rayfire_mesh.h"
+
+// GRINS
+#include "grins/multiphysics_sys.h"
+#include "grins/assembly_context.h"
+#include "grins/materials_parsing.h"
+
+// libMesh
+#include "libmesh/getpot.h"
+#include "libmesh/fem_system.h"
+#include "libmesh/quadrature.h"
+#include "libmesh/point_locator_list.h"
+#include "libmesh/elem.h"
+#include "libmesh/edge_edge2.h"
+#include "libmesh/analytic_function.h"
+#include "libmesh/enum_elem_type.h"
+#include "libmesh/fe.h"
+#include "libmesh/fe_interface.h"
+
+namespace GRINS
+{
+  RayfireMesh::RayfireMesh(libMesh::Point& origin, libMesh::Real theta, libMesh::Real phi) : 
+    _dim(3),_origin(origin),_theta(theta),_phi(phi)
+  {
+    libmesh_not_implemented();
+  }
+ 
+ 
+  RayfireMesh::RayfireMesh(libMesh::Point& origin, libMesh::Real theta) : 
+    _dim(2),_origin(origin),_theta(theta),_phi(0)
```

Comment:
For the 2D case, initialize _phi to some nonsensical value instead of 0 so you can assert later that phi was set to a valid value.